### PR TITLE
IGVF-1539 Audit_category in two places

### PR DESCRIPTION
**Pull request by zhwshen**

### CODE REVIEW

File: src/igvfd/audit/analysis_set.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def audit_analysis_set_samples(value, system):
                         f'Analysis set {audit_link(path_to_text(value["@id"]), value["@id"])} '
                         f'does not specify samples {missing_samples} associated with its `input_file_sets`. '
                     )
-                    yield AuditFailure('missing samples', f'{detail} {missing_description}', level='WARNING')
+                    yield AuditFailure(audit_message_missing_samples.get('audit_category', ''), f'{detail} {audit_message_missing_samples.get("audit_description", "")}', level='WARNING')
```
Comment (ian-whaling) [inline src/igvfd/audit/analysis_set.py:133]: missing assigning the level

File: src/igvfd/audit/analysis_set.py [hunk 2]
```diff
@@ -139,13 +139,13 @@ def audit_analysis_set_samples(value, system):
                         f'Analysis set {audit_link(path_to_text(value["@id"]), value["@id"])} '
                         f'specifies samples {unexpected_samples} not associated with its `input_file_sets`.'
                     )
-                    yield AuditFailure('unexpected samples', f'{detail} {unexpected_description}', level='WARNING')
+                    yield AuditFailure(audit_message_unexpected_samples.get('audit_category', ''), f'{detail} {audit_message_unexpected_samples.get("audit_description", "")}', level='WARNING')
```
Comment (ian-whaling) [inline src/igvfd/audit/analysis_set.py:142]: missing assigning the level

File: src/igvfd/audit/analysis_set.py [hunk 3]
```diff
@@ -139,13 +139,13 @@ def audit_analysis_set_samples(value, system):
...
         else:
             detail = (
                 f'Analysis set {audit_link(path_to_text(value["@id"]), value["@id"])} '
                 f'has `input_file_sets`, but no `samples`.'
             )
-            yield AuditFailure('missing samples', f'{detail} {missing_description}', level='WARNING')
+            yield AuditFailure(audit_message_missing_samples.get('audit_category', ''), f'{detail} {audit_message_missing_samples.get("audit_description", "")}', level='WARNING')
```
Comment (ian-whaling) [inline src/igvfd/audit/analysis_set.py:148]: missing assigning the level

File: src/igvfd/audit/analysis_set.py [hunk 4]
```diff
@@ -172,4 +173,4 @@ def audit_analysis_set_files_missing_analysis_step_version(value, system):
             f'Analysis set {audit_link(path_to_text(value["@id"]), value["@id"])} '
             f'links to file(s) {files_with_missing_asv} that are missing `analysis_step_version`.'
         )
-        yield AuditFailure('missing analysis step version', f'{detail} {description}', level='WARNING')
+        yield AuditFailure(audit_message_missing_step_version.get('audit_category', ''), f'{detail} {audit_message_missing_step_version.get("audit_description", "")}', level='WARNING')
```
Comment (ian-whaling) [inline src/igvfd/audit/analysis_set.py:176]: missing assigning the level

File: src/igvfd/audit/biosample.py [hunk 5]
```diff
@@ -43,7 +43,7 @@ def audit_biosample_taxa_check(value, system):
                 taxa_donors.append(f'{k} ({", ".join(v)})')
             taxa_detail = ', '.join(taxa_donors)
             detail = f'{object_type} {audit_link(path_to_text(sample_id), sample_id)} has `donors` with `taxa` {taxa_detail}. '
-            yield AuditFailure('inconsistent donor taxa', f'{detail} {description}', level='ERROR')
+            yield AuditFailure(audit_message.get('audit_category', ''), f'{detail} {audit_message.get("audit_description", "")}', level='ERROR')
```
Comment (ian-whaling) [inline src/igvfd/audit/biosample.py:46]: missing assigning the level

File: src/igvfd/audit/biosample.py [hunk 6]
```diff
@@ -60,14 +60,14 @@ def audit_biosample_age(value, system):
...
-    description = get_audit_description(audit_biosample_age)
+    audit_message = get_audit_message(audit_biosample_age)
     if 'lower_bound_age' not in value and 'upper_bound_age' not in value and 'age_units' not in value:
         value_id = system.get('path')
         detail = (
             f'{object_type} {audit_link(path_to_text(value_id), value_id)} '
             f'is missing `upper_bound_age`, `lower_bound_age`, and `age_units`.'
         )
-        yield AuditFailure('missing age', f'{detail} {description}', level='WARNING')
+        yield AuditFailure(audit_message.get('audit_category', ''), f'{detail} {audit_message.get("audit_description", "")}', level='WARNING')
```
Comment (ian-whaling) [inline src/igvfd/audit/biosample.py:70]: missing assigning the level

File: src/igvfd/audit/file_set.py [hunk 7]
```diff
@@ -48,7 +48,7 @@ def audit_no_files(value, system):
             f'{object_type} {audit_link(path_to_text(value["@id"]), value["@id"])} '
             f'has no `files`.'
         )
-        yield AuditFailure('missing files', f'{detail} {description}', level='WARNING')
+        yield AuditFailure('missing files', f'{detail} {audit_message.get("audit_description", "")}', level='WARNING')
```
Comment (ian-whaling) [inline src/igvfd/audit/file_set.py:51]: missing assigning the level

**Review by ian-whaling — CHANGES_REQUESTED** [review]
missing assignment of the level in many places

File: src/igvfd/audit/analysis_set.py (hunk 1 above)
Comment (zhwshen) [reply src/igvfd/audit/analysis_set.py:133]: Changed.

File: src/igvfd/audit/analysis_set.py (hunk 2 above)
Comment (zhwshen) [reply src/igvfd/audit/analysis_set.py:142]: Changed.

File: src/igvfd/audit/analysis_set.py (hunk 3 above)
Comment (zhwshen) [reply src/igvfd/audit/analysis_set.py:148]: Changed.

File: src/igvfd/audit/analysis_set.py (hunk 4 above)
Comment (zhwshen) [reply src/igvfd/audit/analysis_set.py:176]: Changed.

File: src/igvfd/audit/biosample.py (hunk 6 above)
Comment (zhwshen) [reply src/igvfd/audit/biosample.py:70]: Addressed.

File: src/igvfd/audit/biosample.py (hunk 5 above)
Comment (zhwshen) [reply src/igvfd/audit/biosample.py:46]: Addressed.

File: src/igvfd/audit/biosample.py [hunk 8]
```diff
@@ -102,4 +102,4 @@ def audit_biomarker_name(value, system):
                     f'{object_type} {audit_link(path_to_text(sample_id), sample_id)} has conflicting biomarkers '
                     f'{biomarkers_to_link} with the same `name`: {name}.'
                 )
-                yield AuditFailure('inconsistent biomarkers', f'{detail} {description}', level='ERROR')
+                yield AuditFailure(audit_message.get('audit_category', ''), f'{detail} {audit_message.get("audit_description", "")}', level='ERROR')
```
[the remark this answers is not in the document]
Comment (zhwshen) [reply src/igvfd/audit/biosample.py:105]: Addressed.

File: src/igvfd/audit/file_set.py (hunk 7 above)
Comment (zhwshen) [reply src/igvfd/audit/file_set.py:51]: Addressed.

File: src/igvfd/audit/file.py [hunk 9]
```diff
@@ -69,4 +69,4 @@ def audit_file_format_specifications(value, system):
                 f'{object_type} {audit_link(path_to_text(value["@id"]), value["@id"])} has `file_format_specification` {audit_link(path_to_text(document), document)} '
                 f'with `document_type` {doc_type}.'
             )
-            yield AuditFailure('inconsistent document type', f'{detail} {description}', level='ERROR')
+            yield AuditFailure('inconsistent document type', f'{detail} {audit_message.get("audit_description", "")}', level=audit_message.get('audit_level', ''))
```
Comment (ian-whaling) [inline src/igvfd/audit/file.py:72]: missing audit category get

File: src/igvfd/tests/test_commands_make_audit_docstring_json.py [hunk 10]
```diff
@@ -19,6 +19,7 @@ def test_get_docstring_dict_from_function_name_docstring_is_defined():
 
 
 def test_get_docstring_dict_from_function_name_docstring_improper_keys():
+    # print(get_docstring_dict_from_function_name('igvfd.tests.fixtures.audit_docstring.function_with_docstring_improper_keys'))
```
Comment (ian-whaling) [inline src/igvfd/tests/test_commands_make_audit_docstring_json.py:22]: remove print statement

**Review by ian-whaling — CHANGES_REQUESTED** [review]
minor comments

**Review by ian-whaling — APPROVED** [review]

**Review by ian-whaling — APPROVED** [review]

**Pull request merged**